### PR TITLE
Provide extra clarity for how to build the docs locally

### LINKED ISSUE #818: How To Build Docs locally is confusing
This document:

https://github.com/owncloud/docs/blob/master/docs/build-the-docs.md

grafik

I want to build an documentation server locally to see if it's possible following our documentation.

I stumble over the first things. My questions:

-  how do I install yarn? 
The wording is very confusing 

> Before you can build the documentation, you need to install the required dependencies. To install them, you need to run yarn install. This will install all the dependencies specified in package.json.

- What is in package.json? Where is this file?
- "To install them, you need to run yarn install." So if I copy paste "yarn install" I will have all the dependencies?

What is GNU? This is a documentation, why are we using abbreviations? This should be a place where you get stuff explained and not the place  were certain knowledge is expected and if you don't have it - go away.

- "you can run make setup instead." Okay, so if I type "run make" then all my dependencies will be installed?

### PR DESCRIPTION
**Pull request by settermjd**
It adds a section showing how to install any prerequisite software and further clarifies how to install Antora's dependencies. It fixes #818.

### CODE REVIEW

**Comment by settermjd** [comment]
@voroyam, can you review, pls?

File: docs/build-the-docs.md [hunk 1]
```diff
@@ -1,15 +1,96 @@
...
+- [Yarn][link-Yarn] (command: `yarn`)
 
-```console
-make setup
+### Checking for the Prerequisites on Linux
+
+To check if they are installed, when running a Linux distribution, run the following command:
+
+```
+dependencies=( git node npm ) && for i in "${dependencies[@]}"; do command -v $i; done;
```
Comment (mmattel) [inline docs/build-the-docs.md]: no yarn?

Comment (settermjd) [reply docs/build-the-docs.md]: Must have missed that.

File: docs/build-the-docs.md [hunk 2]
```diff
@@ -1,15 +1,96 @@
...
 
-## Install Antora's Dependencies
+## Install the Prerequisites
 
-Before you can build the documentation, you need to install the required dependencies. To install them, you need to run `yarn install`. This will install all the dependencies specified in `package.json`.
+Before you can build the ownCloud documentation, you need to install the following software:
 
-**Note:** If your environment already has GNU make installed, you can run `make setup` instead.
+- [git][link-git] (command: `git`)
+- [Node 8][link-Node-8] (command: `node`)
```
Comment (voroyam) [inline docs/build-the-docs.md]: How do you install it? 

if you enter "apt install node"  you get an error.

**Comment by voroyam** [comment]
@settermjd I think the thought behind making a document "How to build the docs locally" is that a user can read it and just with that document at hand get the docs running locally. I feel like you expect existing developer knowledge from the user. 

It would be the best if you just install a fresh 18.04 Ubuntu on a VM and document your process to getting the docs there working. Otherwise it will always be corrupted by the preexisting knowledge / software you already have and others don't.

**Comment by tboerger** [comment]
I don't share the opinion of voroyam. Imho it's totally fine to link to the official install instructions for node, git and so on. We should not document that on our own.

File: docs/build-the-docs.md [hunk 3]
```diff
@@ -1,15 +1,96 @@
...
+- [Gulp CLI][link-Gulp-CLI] (command: `gulp`)
```
Comment (tboerger) [inline docs/build-the-docs.md]: Gulp is not needed for the documentation, it's only needed to build a custom theme.

Comment (tboerger) [reply docs/build-the-docs.md]: Besides that, gulp gets already installed via yarn within the ui repo.

File: docs/build-the-docs.md [hunk 4]
```diff
@@ -1,15 +1,96 @@
...
+nvm use 8
+```
+
+To make Node 8 the default in new terminals, type:
+
+```
+nvm alias default 8
+```
+
+#### Gulp
```
Comment (tboerger) [inline docs/build-the-docs.md]: Drop that entire section. Nobody should install gulp manually.

**Comment by settermjd** [comment]
With respect to https://github.com/owncloud/docs/pull/870#issuecomment-479436902, I agree with @tboerger. The very nature of the web is to be able to link to other, supporting, information. I see no sense in creating — and maintaining — what others already have.

**Comment by voroyam** [comment]
> 
> I don't share the opinion of voroyam. Imho it's totally fine to link to the official install instructions for node, git and so on. We should not document that on our own.

Well, then link it. I just want the commands be there, linked or written down. 

When I tried to follow that document, I clicked on the link and got redirected to some site of the product instead of the instructions.

File: docs/build-the-docs.md (hunk 2 above)
Comment (settermjd) [reply docs/build-the-docs.md]: In the following two sections, there are instructions for testing if they're already installed on both Linux and Windows, and the section after that, there are instructions for installing them.

**Comment by settermjd** [comment]
@voroyam, which link did you click that improperly redirected you?

**Comment by settermjd** [comment]
I've made a couple of further updates, with links to further information, as well as further installation instructions for Windows users.

**Comment by voroyam** [comment]
> @voroyam, which link did you click that improperly redirected you?

No idea, the first one I suppose. To install node. I tried to "apt install node" but it did not work, then I clicked the link for node, and got redirected to the node home page where I could not see how to install it.

I can't show you exactly because the page I was working with does not exist anymore, does it?

**Comment by voroyam** [comment]
> I've made a couple of further updates, with links to further information, as well as further installation instructions for Windows users.

Have you tested those?

**Pull request merged**